### DISCUSSION TITLE
Page numbers / total page count

### DISCUSSION THREAD

**Discussion by patric-r** [eclipse-birt/birt#740] [Q&A]
Maybe I'm wrong, but BIRT has some limitation when using the "autotext" for the page numbers and especially total page count.

We would need to exclude certain pages (or better: elements with span across pages) from the page numbers / total page count in order to avoid ugly workarounds (e.g. combining several pdfs together).

Is there a way to this in BIRT right now and if not, what would needs to be done in order to achieve this?

**Comment by hvbtup** [comment]
I don't think this is possible with BIRT.

If the output format is PDF, then there is workaround that I am using:

Generate a PDF with a PDF TOC with BIRT.

While doing this, add information to the appContext about how to use the TOC entries regarding page numbering and other things like dynamic header or footer texts.

Use iText from within the same Java process to read the information from the appContext and the PDF.

Now the TOC entries can be resolved to physical page numbers.

Use iText to stick the page numbers or other text onto the BIRT-generated PDF pages and save the resulting PDF.

This is ugly, too, but the nice thing about this is that we can work with RunAndRenderTask.
